syntax = "proto3";

// MessageRequest is the request message for the SendMessage method.
// It contains the message and a flag to indicate if the message is from a bot.
message MessageRequest {
    // message is the message to send.
    string message = 1;
    // is_bot is a flag to indicate if the message is from a bot.
    bool is_bot = 2;
}

// MessageResponse is the response message for the SendMessage method from the chat service.
//
// It contains a flag to indicate if the message was successfully sent and an error message if there was an error.
message MessageResponse {
    // success is a flag to indicate if the message was successfully sent.
    bool success = 1;
    // error is the error message if there was an error sending the message, it is empty if the message was successfully sent.
    string error = 2;
    // message is the message that was sent.
    // TODO: Remove this field from the response message.
    string message = 3;
}

service Chat {
    // SendMessage sends a message to the chat service.
    rpc SendMessage(MessageRequest) returns (MessageResponse) {}
}